src/coreclr/jit/importer.cpp

```diff
@@ -8231,6 +8231,35 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
         }
         break;
 
+        case CORINFO_FIELD_STATIC_DATASEGMENT:
+        {
+#ifdef FEATURE_READYTORUN
+            assert(opts.IsReadyToRun());
+            assert((pFieldInfo->fieldFlags & CORINFO_FLG_FIELD_INITCLASS) == 0);
+            
+            if (pFieldInfo->fieldLookup.accessType == InfoAccessType::IAT_VALUE)
+            {
+                op1 = gtNewIconHandleNode((size_t)pFieldInfo->fieldLookup.addr, GTF_ICON_STATIC_HDL);
+                op1->gtType = TYP_BYREF;
+            }
+            else
+            {
+                assert(pFieldInfo->fieldLookup.accessType == InfoAccessType::IAT_PVALUE);
+                op1 = gtNewIndOfIconHandleNode(TYP_BYREF, (size_t)pFieldInfo->fieldLookup.addr, GTF_ICON_GLOBAL_PTR, false);
+            }
+
+            FieldSeqNode* fs = GetFieldSeqStore()->CreateSingleton(pResolvedToken->hField);
+            op1              = gtNewOperNode(GT_ADD, TYP_BYREF, op1,
+                                new (this, GT_CNS_INT) GenTreeIntCon(TYP_INT, pFieldInfo->offset, fs));
```

Comment:
This should use `gtNewIconNode(pFieldInfo->offset, fs)` (that will also use `TYP_I_IMPL` for the offset).

Reply:
I'll update that. It's a copy/paste from the other `case`s in this method.